datachannel.go

```diff
@@ -228,6 +236,35 @@ func (c *DataChannel) MessagesReceived() uint32 {
 	return atomic.LoadUint32(&c.messagesReceived)
 }
 
+// OnOpen sets an event handler which is invoked when a DATA_CHANNEL_ACK message
+// is received or sent on the stream.
+func (c *DataChannel) OnOpen(f func()) {
+	c.mu.Lock()
+	c.openHandlerOnce = sync.Once{}
+	c.onOpenHandler = f
+	open := c.open
+	c.mu.Unlock()
+
+	if open {
```

Comment:
@jwhited sorry about the delay working through this now!

I don't think we should have the `if open` block, I just checked in the browser and `OnOpen` events are not fired when setting the handler late.

Reply:
Beyond this LGTM! Would love to know if there is something I am missing, I can merge and delete this block myself though :)

Reply:
Thanks for taking a look!

I carried this over from pion/webrtc.Datachannel which always fires the handler. Removing this would cause the "Server" side to never fire as there's no way to set the handler before `Server()` returns.